source/m2mnsdlinterface.cpp

```diff
@@ -1649,6 +1641,25 @@ void M2MNsdlInterface::handle_bootstrap_error()
     _observer.bootstrap_error();
 }
 
+void M2MNsdlInterface::set_connection_handler(M2MConnectionHandler *connection_handler)
```

Comment:
Is there really a need to set this variable later, after construction? If the connection_handler was given to constructor and preferably even stored as a reference, there would not be a need to even have those "if (_connection_handler)" checks later.